### PR TITLE
Use RxJava to create a non blocking pipeline to fetch open orders

### PR DESCRIPTION
**Pull request by dazito**
Removed the thread sleep when fetching for open orders.
Built a non blocking pipeline using (RxJava) Observables.

Use a flag to control the flow, if we have open orders then we skip any incoming event. Not very happy with this flag approach so any suggestion is very welcome.

### CODE REVIEW

**Comment by dazito** [comment]
It misses the logging part (calling `collectOpenOrders` and logging it), I will try to add it tomorrow

**Comment by dazito** [comment]
Ready for review

**Comment by scionaltera** [comment]
I also saw the `Trades executed successfully!` message before the trades had both filled when I was testing this overnight. I'm experimenting with it a little bit and I'll keep you posted if I figure anything out. It looks good conceptually though.

File: src/main/java/com/r307/arbitrader/service/TradingService.java [hunk 1]
```diff
@@ -761,47 +771,34 @@ private void executeOrderPair(Exchange longExchange, Exchange shortExchange,
 
         LOGGER.info("Waiting for limit orders to complete...");
 
-        OpenOrders longOpenOrders = null;
-        OpenOrders shortOpenOrders = null;
-        int count = 0;
+        final Observable<OpenOrders> shorOpenOrdersObservable = checkForOpenOrders(shortExchange);
```
Comment (scionaltera) [inline src/main/java/com/r307/arbitrader/service/TradingService.java:774]: `shor -> t <- OpenOrdersObservable` :)

**Comment by dazito** [comment]
Ready to be reviewed :) 
My local tests :heavy_check_mark:

**Review by scionaltera — APPROVED** [review]

**Pull request merged**